pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -312,22 +314,29 @@ func (c *ReverseExpandQuery) execute(
 			}
 			err = c.execute(ctx, r, resultChan, intersectionOrExclusionInPreviousEdges, resolutionMetadata)
 			if err != nil {
-				return err
+				errs = multierror.Append(errs, err)
+				break LoopOnEdges
 			}
 		case graph.TupleToUsersetEdge:
 			pool.Go(func(ctx context.Context) error {
 				return c.reverseExpandTupleToUserset(ctx, r, resultChan, intersectionOrExclusionInPreviousEdges, resolutionMetadata)
 			})
 		default:
-			return fmt.Errorf("unsupported edge type")
+			errs = multierror.Append(errs, fmt.Errorf("unsupported edge type"))
```

Comment:
I think this should be a panic. The `switch` already covers all possible graph edge types, if we messed this up then something went very wrong.